### PR TITLE
docs: restructure scanners

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
- Move all scanners under `Scanner`
- Move common options under `Configuration`
- Move SBOM and attestation under `Supply Chain`

### Before
image

### After
image

## Issues
- Close https://github.com/aquasecurity/trivy/issues/3621

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: docs/docs/configuration/reporting.md [hunk 1]
```diff
@@ -1,13 +1,34 @@
-# Report Formats
+# Reporting
+
+Trivy supports the following formats:
+
+- table
+- json
+- sarif 
+- template
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply docs/docs/configuration/reporting.md]: Nice catch. Done.
https://github.com/aquasecurity/trivy/pull/3977/commits/328bdcdbe1737ea3853322327caff6fc82cd773b

File: docs/docs/configuration/reporting.md [hunk 2]
```diff
@@ -273,15 +315,35 @@ $ trivy image --format template --template "@/path/to/template" golang:1.12-alpi
 If Trivy is installed using rpm then default templates can be found at `/usr/local/share/trivy/templates`.
 
 #### XML
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply docs/docs/configuration/reporting.md:315]: The title was wrong. It should have been `JUnit` here.

[the remark this answers is not in the document]
Comment (knqyf263) [reply docs/docs/configuration/reporting.md]: Fixed in https://github.com/aquasecurity/trivy/pull/3977/commits/5a3602d78d2cf3bf95030109e40947723218039f

File: docs/docs/configuration/reporting.md [hunk 3]
```diff
@@ -224,6 +259,13 @@ This SARIF file can be uploaded to GitHub code scanning results, and there is a
...
 ## Template
 
+|     Scanner      | Supported |
+|:----------------:|:---------:|
+|  Vulnerability   |     ✓     |
+| Misconfiguration |     ✓     |
+|      Secret      |     ✓     |
+|     License      |     ✓     |
+
 ### Custom Template
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply docs/docs/configuration/reporting.md:267]: JUnit was added.
https://github.com/aquasecurity/trivy/pull/3977/commits/5a3602d78d2cf3bf95030109e40947723218039f

I want to deprecate GitLab templates, so I didn't add them.
https://github.com/aquasecurity/trivy/issues/4033

File: docs/docs/configuration/others.md [hunk 4]
```diff
@@ -37,8 +68,15 @@ Will skip all subdirectories of the testdata directory.
 
 
 ## File patterns
+|     Scanner      | Supported |
+|:----------------:|:---------:|
+|  Vulnerability   |           |
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply docs/docs/configuration/others.md]: Nice catch! I forgot about it 😆 
https://github.com/aquasecurity/trivy/pull/3977/commits/c6c75f27b7733b9d17eb33caccf4a2f701ee0022

**Review by DmitriyLewen — APPROVED** [review]

File: examples/misconf/custom-policy/terraform-plan/README.md [hunk 5]
```diff
@@ -1,5 +0,0 @@
-# Terraform Plan
-
-```
-$ trivy conf --policy ./policy --namespaces user ./configs
-```
```
Comment (simar7) [inline examples/misconf/custom-policy/terraform-plan/README.md:5]: @knqyf263 - was this moved somewhere else? I am unable to find it anywhere. As a result it has broken most of the links on this page: https://aquasecurity.github.io/trivy/v0.41/docs/scanner/misconfiguration/custom/examples/

**Pull request merged**